staging/src/k8s.io/apiserver/pkg/server/options/encryptionconfig/config_test.go

```diff
@@ -721,6 +731,83 @@ func TestKMSv2PluginHealthzTTL(t *testing.T) {
 	}
 }
 
+func TestKMSv2InvalidKeyID(t *testing.T) {
+	ctx := testContext(t)
+	invalidKeyIDService, _ := newMockInvalidKeyIDEnvelopeKMSv2Service(ctx, "unix:///tmp/testprovider.sock", "providerName", 3*time.Second, "")
+	service, _ := newMockInvalidKeyIDEnvelopeKMSv2Service(ctx, "unix:///tmp/testprovider.sock", "providerName", 3*time.Second, "1")
+
+	testCases := []struct {
+		desc    string
+		probe   *kmsv2PluginProbe
+		metrics []string
+		want    string
+	}{
+		{
+			desc: "kmsv2 provider returns an invalid keyID",
+			probe: &kmsv2PluginProbe{
+				name:         "test",
+				ttl:          kmsPluginHealthzNegativeTTL,
+				service:      invalidKeyIDService,
+				l:            &sync.Mutex{},
+				lastResponse: &kmsPluginHealthzResponse{},
+			},
+			metrics: []string{
+				"apiserver_envelope_encryption_invalid_key_id_from_status_total",
+			},
+			want: `
+			# HELP apiserver_envelope_encryption_invalid_key_id_from_status_total [ALPHA] Number of times an invalid keyID is returned by the Status RPC call.
+			# TYPE apiserver_envelope_encryption_invalid_key_id_from_status_total counter
+			apiserver_envelope_encryption_invalid_key_id_from_status_total{provider_name="test"} 1
+			`,
+		},
+		{
+			desc: "kmsv2 provider returns a valid keyID",
+			probe: &kmsv2PluginProbe{
+				name:         "test",
+				ttl:          kmsPluginHealthzNegativeTTL,
+				service:      service,
+				l:            &sync.Mutex{},
+				lastResponse: &kmsPluginHealthzResponse{},
+			},
+			metrics: []string{
+				"apiserver_envelope_encryption_invalid_key_id_from_status_total",
+			},
+			want: ``,
+		},
+		{
+			desc: "kmsv2 provider returns an invalid keyID",
+			probe: &kmsv2PluginProbe{
+				name:         "test",
+				ttl:          kmsPluginHealthzNegativeTTL,
+				service:      invalidKeyIDService,
+				l:            &sync.Mutex{},
+				lastResponse: &kmsPluginHealthzResponse{},
+			},
+			metrics: []string{
+				"apiserver_envelope_encryption_invalid_key_id_from_status_total",
+			},
+			want: `
+			# HELP apiserver_envelope_encryption_invalid_key_id_from_status_total [ALPHA] Number of times an invalid keyID is returned by the Status RPC call.
+			# TYPE apiserver_envelope_encryption_invalid_key_id_from_status_total counter
+			apiserver_envelope_encryption_invalid_key_id_from_status_total{provider_name="test"} 1
+			`,
+		},
```

Comment:
duplicated test? (ref: https://github.com/kubernetes/kubernetes/pull/115846/files#diff-f2f067d6aa03c572d14285964d389066c885f5f54df7dc7d779ccecdc50f7b75R746-R762)

Reply:
not duplicate. I wanted to see the metric go from 1 item to empty, then to 1 item.

Reply:
I updated this test to show another error instead so it doesn't look like a dup.